### DISCUSSION TITLE
Some proposals

### DISCUSSION THREAD

**Discussion by ghost** [Ahwxorg/LibreY#166] [Ideas]
Hi, first of all I compliment the team for this interesting project, I have some things that I’d like to be implemented if possible:
1) are you planning on adding more engines to the image search other than qwant?
2) can you make the public instances’ fallback optional? Maybe as a UI setting? That could make sense for any user who just wants to rely on his own local instance.
3) when I choose an engine and search, switching to another engine requires going back to the settings and save. Is it possible to make the switch faster? For example, 4get has a drop-down menu on top of the results page where you can select another engine and you get an instantaneous search with the scraper that you’ve chosen.
 Thanks in advance.

**Comment by Ahwxorg** [comment]
> are you planning on adding more engines to the image search other than qwant?

Not planning, but a PR would definitely be welcome!

> can you make the public instances’ fallback optional? Maybe as a UI setting? That could make sense for any user who just wants to rely on his own local instance.

It actually is optional! The host can set that for now, but I could definitely see some use in the user being able to set that as well.

> when I choose an engine and search, switching to another engine requires going back to the settings and save. Is it possible to make the switch faster? For example, 4get has a drop-down menu on top of the results page where you can select another engine and you get an instantaneous search with the scraper that you’ve chosen.

Should be possible, but I don't think I'd like that to be fair; mostly due to rate-limiting.
